packages/upload-api/src/types.ts

```diff
@@ -403,6 +403,7 @@ export type UploadServiceContext = ConsumerServiceContext &
   }
 
 export interface AccessClaimContext {
+  signer: EdSigner.Signer
```

Comment:
Can just be `principal.Signer` no? No need to be ed25519?

Reply:
Agreed. They were all this, but I've changed them to just `Signer` now, which appears to be the right one. There are a lot of things called "Signer", though 😅